src/event-stream/event-server.ts

```diff
@@ -236,6 +237,17 @@ async function handleClientMessage(
         abi: JSON.stringify(tx.core_tx.contract_abi),
         canonical: true,
       });
+      //check if this contract uses fungible/non fungible tokens
+      if (tx.core_tx.contract_abi && hasTokens(tx.core_tx.contract_abi)) {
+        //TODO start it in a seperate thread
+        await new TokensContractHandler(
```

Comment:
It looks like this could deadlock the event handler. If the stacks-node is waiting for an http response to the `/new_block` event request, then the stacks-node will likely not respond to the `/v2/...` requests made during this call.

We should setup a promise queue that is set to evaluate 1 of these at a time. The lib `p-queue` is useful for that, and already being used, see example here:
https://github.com/blockstack/stacks-blockchain-api/blob/9ec64ad13403a7e82e6aa5d3648a17cb3133c809/src/event-stream/event-server.ts#L532